### PR TITLE
Initial migration of codebase

### PR DESCRIPTION
**Pull request by digitaldan**
Signed-off-by: Dan Cunningham <dan@digitaldan.com>

### CODE REVIEW

**Comment by digitaldan** [comment]
Not sure why I closed it!

**Comment by kaikreuzer** [comment]
@digitaldan Please add "Also-by: ..." footers for all authors that worked on this code.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+We recommend discussing your plans [in the discussion forum](https://community.openhab.org/)
+before starting to code - especially for more ambitious contributions.
+This gives other contributors a chance to point you in the right
+direction, give feedback on your design, and maybe point out if someone
+else is working on the same thing.
+
+### Create issues...
+
+Any significant improvement should be documented as [a GitHub
+issue](https://github.com/openhab/openhab-docker/issues?labels=enhancement&page=1&state=open) before anybody
```
Comment (kaikreuzer) [inline CONTRIBUTING.md]: should be openhab-cloud, not openhab-docker!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+We are always thrilled to receive pull requests, and do our best to
+process them as fast as possible. Not sure if that typo is worth a pull
+request? Do it! We will appreciate it.
+
+If your pull request is not accepted on the first try, don't be
+discouraged! If there's a problem with the implementation, hopefully you
+received feedback on what to improve.
+
+We're trying very hard to keep openHAB-cloud lean and focused. We don't want it
```
Comment (kaikreuzer) [inline CONTRIBUTING.md]: openHAB-cloud -> the openHAB Cloud

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+to do everything for everybody. This means that we might decide against
+incorporating a new feature. However, there might be a way to implement
+that feature *on top of* openHAB-cloud.
```
Comment (kaikreuzer) [inline CONTRIBUTING.md]: openHAB-cloud -> the openHAB Cloud

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+#### Small patch exception
+
+There are several exceptions to the signing requirement. Currently these are:
+
+* Your patch fixes spelling or grammar errors.
+* Your patch is a single line change to documentation.
+
+## Community Guidelines
+
+We want to keep the openHAB-cloud community awesome, growing and collaborative. We
```
Comment (kaikreuzer) [inline CONTRIBUTING.md]: openHAB-cloud -> openHAB

File: config-development.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+    "system": {
+      "logging": "debug",
+      "baseurl": "openhab-cloud.org:443"
```
Comment (kaikreuzer) [inline config-development.json]: what basurl is that? If it is meant as a placeholder, better mark it as such (i.e. `<your-openhab-cloud-host>:443`)

File: config-development.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        "port": "6379",
+        "password": "secret"
+    },
+    "mailer": {
+        "host" : "somehost",
+        "port": 465,
+        "secureConnection": true,
+        "user": "someuser",
+        "password": "somepass",
+        "from": "My openHAB <my@openhab.org>"
```
Comment (kaikreuzer) [inline config-development.json]: change to "`openHAB Cloud <your@email.address>`"

File: config-production.json [hunk 7]
```diff
@@ -0,0 +1,48 @@
+{
+    "system": {
+      "baseurl": "myopenhab.org",
```
Comment (kaikreuzer) [inline config-production.json]: This should again be `<your-openhab-cloud-host>:443`, right?

File: config-production.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        "port": "6379",
+        "password": "password"
+    },
+    "mailer": {
+        "host" : "smtp",
+        "port": 465,
+        "secureConnection": true,
+        "user": "foo@bar.com",
+        "password": "password",
+        "from": "My openHAB <my@openhab.org>"
```
Comment (kaikreuzer) [inline config-production.json]: change to "`openHAB Cloud <your@email.address>`"

File: etc/nginx_openhabcloud.conf [hunk 9]
```diff
@@ -0,0 +1,64 @@
+server {
+listen *:443;
+listen *:80;
+ssl on;
+ssl_certificate /etc/nginx/ssl/myopenhab.crt;
```
Comment (kaikreuzer) [inline etc/nginx_openhabcloud.conf]: openhabcloud.crt?

File: etc/nginx_openhabcloud.conf [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+ssl_certificate_key /etc/nginx/ssl/myopenhab.key;
```
Comment (kaikreuzer) [inline etc/nginx_openhabcloud.conf]: openhabcloud.key?

File: etc/nginx_openhabcloud.conf [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+server_name openhab-cloud.org;
```
Comment (kaikreuzer) [inline etc/nginx_openhabcloud.conf]: This doesn't exist. Should it be again `<your-openhab-cloud-host>`?

File: models/oauth2code.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+var mongoose = require('mongoose'),
+    Schema = mongoose.Schema;
+var ObjectId = mongoose.SchemaTypes.ObjectId;
+
+var OAuth2CodeSchema = new Schema({
+    user: ObjectId,                             // my.OH user this code belongs to
```
Comment (kaikreuzer) [inline models/oauth2code.js]: openhab-cloud user

File: models/oauth2code.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    oAuthClient: ObjectId,                      // my.OH OAuth2 Client this code was created for
```
Comment (kaikreuzer) [inline models/oauth2code.js]: openhab-cloud OAuth2 Client

File: models/oauth2token.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
+var mongoose = require('mongoose'),
+    Schema = mongoose.Schema;
+var ObjectId = mongoose.SchemaTypes.ObjectId;
+
+var OAuth2TokenSchema = new Schema({
+    user: ObjectId,                             // my.OH user this token belongs to
```
Comment (kaikreuzer) [inline models/oauth2token.js]: openhab-cloud user

File: models/oauth2token.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+    oAuthClient: {type: ObjectId, ref: 'OAuth2Client'}, // my.OH OAuth2 Client this token was created for
```
Comment (kaikreuzer) [inline models/oauth2token.js]: openhab-cloud OAuth2 Client

**Comment by kaikreuzer** [comment]
Wow, you are incredibly quick!

**Pull request merged**